packages/components/src/Loader/Loader.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { PropsWithChildren, RefObject } from "react";
+import classnames from "classnames";
+import { Breakpoints, useResizeObserver } from "@jobber/hooks/";
+import styles from "./Loader.css";
+import { LoaderIndicator } from "./LoaderIndicator";
+
+type LoaderProps = IndeterminateLoaderProps | DeterminateLoaderProps;
+
+interface BaseLoaderProps {
+  /**
+   * Removes the overlay and has the component render in the flow of the layout
+   * @default false
+   */
+  readonly inline?: boolean;
+
+  /**
+   * Shows a spinner when false, progress bar when true.
+   * @default false
+   */
+  determinate?: boolean;
+
+  /**
+   * Show loading wrapper if underlying content is loading.
+   * @default false
+   */
+  loading: boolean;
+}
+
+interface IndeterminateLoaderProps extends BaseLoaderProps {
+  determinate: false;
+}
+
+interface DeterminateLoaderProps extends BaseLoaderProps {
+  determinate: true;
+
+  /**
+   * The determinate loading progress
+   * @default 0
+   */
+  currentValue: number;
+
+  /**
+   * The determinate loading value when loading is completed
+   * @default 100
+   */
+  maxValue?: number;
+}
+
+export function Loader(props: PropsWithChildren<LoaderProps>) {
+  const {
+    inline = false,
+    determinate = false,
+    children,
+    loading = false,
+  } = props;
+  let currentValue, maxValue;
+  if (props.determinate) {
+    currentValue = props.currentValue || 0;
+    maxValue = props.maxValue || 100;
+  }
+
+  const [ref, { width = 0 }] = useResizeObserver();
+  const classname = classnames(styles.loader, {
+    [styles.overlay]: !inline,
+    [styles.determinate]: determinate,
+  });
+
+  return (
+    <div ref={ref as RefObject<HTMLDivElement>}>
+      {loading ? (
+        <div aria-busy="true" aria-live="polite" className={classname}>
+          {children}
+          <div>
```

Comment:
Can we give this div a classname so that we don't need to do the nested selectors in the CSS? :last-child type selectors are quite brittle

Reply:
Added class name in ffa84d8